modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinServerCodegen.java

```diff
@@ -381,6 +396,291 @@ public static class Constants {
+                            }
+                            for (CodegenProperty prop : model.getVars()) {
+                                if (parentPropNames.contains(prop.getBaseName())) {
+                                    prop.isInherited = true;
+                                }
+                            }
+                            for (CodegenProperty prop : model.getRequiredVars()) {
+                                if (parentPropNames.contains(prop.getBaseName())) {
+                                    prop.isInherited = true;
+                                }
+                            }
+                            for (CodegenProperty prop : model.getOptionalVars()) {
+                                if (parentPropNames.contains(prop.getBaseName())) {
+                                    prop.isInherited = true;
+                                }
+                            }
+
+                            // Set vendor extension for parent constructor call with inherited properties
+                            if (!inheritedPropNamesList.isEmpty()) {
+                                String parentCtorArgs = String.join(", ", inheritedPropNamesList.stream()
+                                        .map(name -> name + " = " + name)
+                                        .toArray(String[]::new));
+                                model.getVendorExtensions().put("x-parent-ctor-args", parentCtorArgs);
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Third pass: set vendor extension for discriminator style and handle fixJacksonJsonTypeInfoInheritance
+            for (String ownerName : trueDiscriminatorOwners) {
+                CodegenModel owner = allModelsMap.get(ownerName);
+                if (owner != null && owner.getDiscriminator() != null) {
+                    String discriminatorPropBaseName = owner.getDiscriminator().getPropertyBaseName();
+                    boolean isOneOfOrAnyOfPattern = (owner.oneOf != null && !owner.oneOf.isEmpty())
+                            || (owner.anyOf != null && !owner.anyOf.isEmpty());
+
+                    // hasParentProperties controls whether the sealed class has properties in its constructor
+                    // This should be false for oneOf/anyOf patterns (parent is a type union, no direct properties)
+                    // and true for allOf patterns (parent has properties that children inherit)
+                    boolean hasParentProperties = !isOneOfOrAnyOfPattern;
+
+                    // visibleTrue controls whether visible=true is set on @JsonTypeInfo
+                    // When fixJacksonJsonTypeInfoInheritance is true, we always set visible=true
+                    // When false, we only set visible=true if the parent has properties (allOf pattern)
+                    boolean visibleTrue;
+
+                    if (fixJacksonJsonTypeInfoInheritance) {
```

Comment:
This is the actual implementation of the Jackson-specific fix as mentioned above.